storage/src/vespa/storage/distributor/bucketdbupdater.cpp

```diff
@@ -113,25 +121,35 @@ void
 BucketDBUpdater::removeSuperfluousBuckets(
         const lib::ClusterStateBundle& newState)
 {
+    const bool move_to_read_only_db = shouldDeferStateEnabling();
```

Comment:
After pondering (and discussing) this for a bit, I think it semantically should suffice to only consider the config here. The distributor config is explicitly just for enabling stale reads, and this will work independently of deferred activation. It just won't be coordinated across the cluster.